A50-xds-outlier-detection.md

```diff
@@ -0,0 +1,262 @@
+    // interval (as defined by the interval duration above) to include this address
+    // in success rate based outlier detection. If the volume is lower than this
+    // setting, outlier detection via success rate statistics is not performed
+    // for that address. Defaults to 100.
+    google.protobuf.UInt32Value request_volume = 4;
+  }
+
+  // Parameters for the failure percentage algorithm.
+  // This algorithm ejects individual endpoints whose failure rate is greater than
+  // some threshold, independently of any other endpoint.
+  message FailurePercentageEjection {
+    // The failure percentage to use when determining failure percentage-based outlier detection. If
+    // the failure percentage of a given address is greater than or equal to this value, it will be
+    // ejected. Defaults to 85.
+    google.protobuf.UInt32Value threshold = 1;
+
+    // The % chance that an address will be actually ejected when an outlier status is detected through
+    // failure percentage statistics. This setting can be used to disable ejection or to ramp it up
+    // slowly. Defaults to 100.
+    google.protobuf.UInt32Value enforcement_percentage = 2;
+
+    // The minimum number of addresses in order to perform failure percentage-based ejection.
+    // If the total number of addresses is less than this value, failure percentage-based
+    // ejection will not be performed. Defaults to 5.
+    google.protobuf.UInt32Value minimum_hosts = 3;
+
+    // The minimum number of total requests that must be collected in one interval (as defined by the
+    // interval duration above) to perform failure percentage-based ejection for this address. If the
+    // volume is lower than this setting, failure percentage-based ejection will not be performed for
+    // this host. Defaults to 50.
+    google.protobuf.UInt32Value request_volume = 4;
+  }
+
+  // If set, success rate ejections will be performed
+  SuccessRateEjection success_rate_ejection = 5;
+  
+  // If set, failure rate ejections will be performed
+  FailurePercentageEjection failure_percentage_ejection = 6;
+
+  // The config for the child policy
+  repeated LoadBalancingConfig child_policy = 13;
+}
+```
+
+### Outlier Detection Behavior
+
+Internally, the `outlier_detection` LB policy will store a map with keys that are subchannel addresses, and values that are objects that contain the following:
+
+ - The call result counter object (see [Call Counter section](#call-counter))
+ - The latest ejection timestamp, or `null` if the address is currently not ejected
+ - The current ejection time multiplier, starting at 0
+ - A list of subchannel wrapper objects that correspond to this address
+
+When the `outlier_detection` LB policy receives an address update, it will create a map entry for each subchannel address in the list, and remove each map entry for a subchannel address not in the list, then pass the address list along to the subchannel.
```

Comment:
Is the final word "subchannel" supposed to be "subpolicy"?

Reply:
Fixed.